### DISCUSSION TITLE
Working with Barcode binding Magick

### DISCUSSION THREAD

**Discussion by andrermartins** [micjahn/ZXing.Net#485] [General]
Hello,

Someone could help me? I need a implementation example of how to instance `new BarcodeWriter<TQuantumType>{...}` using Magick binding (ZXing.Magick). A example of TQuantumType to put there.

Thank you,

**Comment by micjahn** [comment]
The following snippet should work:

```
var writer = new ZXing.Magick.BarcodeWriter<System.Byte>(new ImageMagick.MagickImageFactory())
{
    Format = BarcodeFormat.QR_CODE,
    Options = new QrCodeEncodingOptions
    {
        // ...
    }
};
var imageMagick = writer.Write("content");
```

**Reply by andrermartins** [reply]
Thank you so much for reply,

I tested this snippet, but the `new MagickImageFactory()` was not found. Do you know what `using` to use?

Thank you,

**Reply by micjahn** [reply]
ImageMagick.MagickImageFactory
from nuget package Magick.NET-Q8-AnyCPU

**Reply by andrermartins** [reply]
Thank you a lot!
